packages/core/utils/lib/convert-query-params.js

```diff
@@ -250,39 +251,52 @@ const sanitizeFilters = (filters, schema) => {
 
   // Here, `key` can either be an operator or an attribute name
   for (const [key, value] of Object.entries(filters)) {
-    const attribute = schema.attributes[key];
+    const attribute = get('key', schema.attributes);
 
     // Handle attributes
     if (attribute) {
-      // Always remove password attributes from filters object
-      if (attribute.type === 'password') {
-        removeOperator(key);
-      }
-
       // Relations
       if (attribute.type === 'relation') {
-        filters[key] = sanitizeFilters(value, strapi.getModel(attribute.target));
+        filters[key] = convertAndSanitizeFilters(value, strapi.getModel(attribute.target));
       }
 
       // Components
       else if (attribute.type === 'component') {
-        filters[key] = sanitizeFilters(value, strapi.getModel(attribute.component));
+        filters[key] = convertAndSanitizeFilters(value, strapi.getModel(attribute.component));
       }
 
       // Media
       else if (attribute.type === 'media') {
-        filters[key] = sanitizeFilters(value, strapi.getModel('plugin::upload.file'));
+        filters[key] = convertAndSanitizeFilters(value, strapi.getModel('plugin::upload.file'));
       }
 
       // Dynamic Zones
       else if (attribute.type === 'dynamiczone') {
         removeOperator(key);
       }
+
+      // Scalar attributes
+      else {
+        // Always remove password attributes from filters object
+        if (attribute.type === 'password') {
+          removeOperator(key);
+        } else {
+          filters[key] = convertAndSanitizeFilters(value);
+        }
+      }
     }
 
     // Handle operators
-    else if (isObject(value)) {
-      filters[key] = sanitizeFilters(value, schema);
+    else {
+      if (isObject(value)) {
+        filters[key] = convertAndSanitizeFilters(value, schema);
+      } else if (['$null', '$notNull'].includes(key)) {
+        try {
+          filters[key] = parseType({ type: 'boolean', value: filters[key], forceCast: true });
+        } catch (e) {
+          throw new ApplicationError(e.message);
```

Comment:
WHat do you think of using a more details error format like we do for Validation Error ? We could have QueryError or similar ?

Reply:
I saw that the error handling can be improved, for example there is already a 500 error when we select a field that doesn't exist.
I highly hesitated to catch the error or to just cast any value there to Boolean. In anycase it doesn't fit a proper error handling system. It would be nice to have one, I thought that we could do it in another US and that we should maybe discuss about it before. What do you think ?

Reply:
Let's discuss it during our next sync :)